addons/sourcemod/scripting/include/json/object.inc

```diff
@@ -106,7 +106,7 @@ methodmap JSON_Object < MetaStringMap
      */
     public int Iterate()
     {
-        if (! this.OrderedKeys) {
+        if (! this.OrderedKeys && ! this.IsArray) {
```

Comment:
This is not technically necessary as we don't call `Iterate` on arrays at all now (I changed it to always use `Length`), but in a forward, someone could still call `Iterate` on an array and create snapshot leaks that way.